app/grandchallenge/components/models.py

```diff
@@ -565,9 +570,32 @@ def file_mimetypes(self):
             )
         elif self.kind == InterfaceKind.InterfaceKindChoices.MP4:
             return ("video/mp4",)
+        elif self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
+            return ("text/x-nh", "application/octet-stream")
         else:
             raise RuntimeError(f"Unknown kind {self.kind}")
 
+    @property
+    def file_extensions(self):
+        exts = [self.file_extension]
+
+        if self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
+            exts.extend([".nwk", ".tree"])
```

Comment:
Does that mean we store data with multiple different extensions? When they're used in a component job is only one file extension used?

Reply:
One is the primary, used to validate the relative path of interfaces. The multiple is the one for uploading. I think it's best to rename the `file_extensions` to `upload_file_extensions` and add some docstrings to explain their function.

Reply:
Indeed, only a single extension in the relative paths of the NEWICK kind. The additional upload extensions should only result in additional extensions in the original filename in the serialization of the CIV. AFAIK that is only applicable when the original uploaders are curious about what input was used to generate component job outputs: and it sort of make sense to keep the original extension there.